casanovo/data/ms_io.py

```diff
@@ -143,10 +230,95 @@ def set_ms_run(self, peak_filenames: List[str]) -> None:
             )
             self._run_map[filename] = i
 
+    def get_report_dict(self) -> Optional[Dict]:
+        """
+        Generate sequencing run report
+
+        Parameters
+        ----------
+            score_bins: List[float], Optional
+                Confidence scores for creating confidence CMF, see getScoreBins
+
+        Returns:
+            report_gen: Dict
+                Generated report represented as a dictionary, or None if no
+                sequencing predictions were logged
+        """
+        results_table = DataFrame(
+            {
+                "sequence": [psm[0] for psm in self.psms],
+                "score": [psm[2] for psm in self.psms],
+            }
+        )
+
+        if results_table.empty:
+            return None
+
+        peptide_lengths = get_peptide_lengths(results_table)
+        return {
+            "num_spectra": len(results_table),
+            "score_bins": get_score_bins(results_table, self.score_bins),
+            "max_sequence_length": int(np.max(peptide_lengths)),
+            "min_sequence_length": int(np.min(peptide_lengths)),
+            "median_sequence_length": int(np.median(peptide_lengths)),
+            "peptide_length_histogram": get_peptide_length_histo(
+                peptide_lengths
+            ),
+        }
+
+    def log_run_report(self) -> None:
+        """
+        Log sequencing run report
+        """
+        logger.info("======= Sequencing Run Report =======")
+        if self.start_time is not None:
+            end_time = time()
+            elapsed_time = end_time - self.start_time
+            logger.info(
+                f"Sequencing Run Start Timestamp: {int(self.start_time)}s"
```

Comment:
**todo:** The timestamps are not really interpretable. I'd convert this to human-readable strings.